sdk/eventhub/Azure.Messaging.EventHubs/src/Producer/EventHubBufferedProducerClient.cs

```diff
@@ -2151,6 +2221,39 @@ private async Task RunProducerManagementAsync(CancellationToken cancellationToke
                 }
             }, cancellationToken);
 
+        /// <summary>
+        ///    Ensures that class state has accurate partition information, updating it if the
+        ///    set of Event Hub partitions has changed since they were last queried.
+        /// </summary>
+        ///
+        /// <param name="cancellationToken">A <see cref="CancellationToken" /> instance to signal the request to cancel the operation.</param>
+        ///
+        /// <remarks>
+        ///   This method will potentially modify class state, overwriting the tracked set of partitions.
+        /// </remarks>
+        ///
+        private async Task UpdatePartitionInformation(CancellationToken cancellationToken)
+        {
+            var currentPartitions = _partitions;
+            var queriedPartitions = await _producer.GetPartitionIdsAsync(cancellationToken).ConfigureAwait(false);
+
+            // Assume that if the count of partitions matches the current count, then no updated is needed.
+            // This is safe because partition identifiers are stable and new partitions can be added but
+            // the can never be removed.
+
+            if ((currentPartitions?.Length ?? 0) != queriedPartitions.Length)
+            {
+                // The partitions need to be updated.  Because the two class members tracking partitions
+                // are used for different purposes, it is permissible for them to drift for a short time.
+                // As a result, there's no need to synchronize them.
+
+                var queriedHash = new HashSet<string>(queriedPartitions);
```

Comment:
nit: is there any value in storing this as a local variable rather than just assigning it to `_partitionHash`?